core/src/main/java/org/jruby/ext/ripper/StringTerm.java

```diff
@@ -69,11 +78,30 @@ private int endFound(RipperLexer lexer) throws IOException {
                 return ' ';
             }
 
-            if ((flags & STR_FUNC_REGEXP) != 0) return parseRegexpFlags(lexer);
+            if ((flags & STR_FUNC_REGEXP) != 0) {
+                validateRegexp(lexer);
+                return RubyParser.tREGEXP_END;
+            }
 
             return RubyParser.tSTRING_END;
     }
 
+    private void validateRegexp(RipperLexer lexer) throws IOException {
+        Ruby runtime = lexer.getRuntime();
+        RegexpOptions options = lexer.parseRegexpFlags();
+        ByteList last = null;
+        for (ByteList fragment : regexpFragments) {
+            last = fragment;
+            if (fragment != null) {
+                lexer.checkRegexpFragment(runtime, fragment, options);
+            }
+        }
+        if (last != null && regexpFragments.size() == 1) {
```

Comment:
Is this only possible for /#{ffofofofo}/ and /@{gfogfogogog}/?  If so can you add a comment here as to what scenario this is for?  Without printing out the lex stream I would almost think this was not possible.  A comment will help later on.

Reply:
Rewrote the implementation with an additional `regexpDynamic` variable instead to make the logic simpler to follow instead on commenting on the cryptic logic.